src/Vault.sol

```diff
@@ -0,0 +1,207 @@
+    function decimals() external view returns (uint);
+}
+
+contract Vault is Ownable {
+
+    ///@dev Vault records collateral deposits of each user
+    mapping (address => uint) public deposits; 
+
+    ///@dev Vault records debt holdings of each user
+    mapping (address => uint) public debts;  
+
+    ///@dev ERC20 interface specifying token contract functions
+    IERC20Decimals public immutable collateral;    
+    IERC20Decimals public immutable debt;  
+
+    ///@dev Asset Pricefeed interface from Chainlink
+    AggregatorV3Interface public immutable priceFeed;   
+    
+    ///@dev Emitted on deposit()
+    ///@param collateralAsset The address of the collateral asset
+    ///@param user The address of the user calling deposit()
+    ///@param collateralAmount The amount of collateral asset deposited
+    event Deposit(address indexed collateralAsset, address indexed user, uint collateralAmount);  
+
+    ///@dev Emitted on borrow()
+    ///@param debtAsset The address of the debt asset
+    ///@param user The address of the user calling borrow()
+    ///@param debtAmount The amount of debt asset borrowed
+    event Borrow(address indexed debtAsset, address indexed user, uint debtAmount); 
+
+    ///@dev Emitted on repay()
+    ///@param debtAsset The address of the debt asset
+    ///@param user The address of the user calling repay()
+    ///@param debtAmount The amount of debt asset being repaid
+    event Repay(address indexed debtAsset, address indexed user, uint debtAmount);
+
+    ///@dev Emitted on withdraw()
+    ///@param collateralAsset The address of the debt asset
+    ///@param user The address of the user calling withdraw()
+    ///@param collateralAmount The amount of collateral asset withdrawn
+    event Withdraw(address indexed collateralAsset, address indexed user, uint collateralAmount);  
+
+    ///@dev Emitted on liquidation()
+    ///@param collateralAsset The address of the debt asset
+    ///@param debtAsset The address of the debt asset
+    ///@param user The address of the user calling withdraw()
+    ///@param debtToCover The amount of debt the liquidator wants to cover
+    ///@param liquidatedCollateralAmount The amount of collateral received by the liquidator
+    event Liquidation(address indexed collateralAsset, address indexed debtAsset, address indexed user, uint debtToCover, uint liquidatedCollateralAmount);
+    
+    ///@dev Returns decimal places of price as dictated by Chainlink Oracle
+    uint public immutable scalarFactor;
+    
+    ///@dev Returns decimal places of tokens as dictated by their respective contracts
+    uint public collateralDecimals;
+    uint public debtDecimals;
+
+    
+    constructor(address dai_, address weth_, address priceFeedAddress) {
+        collateral = IERC20Decimals(weth_);
+        debt = IERC20Decimals(dai_);
+
+        priceFeed = AggregatorV3Interface(priceFeedAddress);
+        scalarFactor = 10**priceFeed.decimals();
```

Comment:
maybe call this priceFeedDecimals (or feedDecimals) keeping w the naming conventions above.  its a bit confusing